src/AdoNet/Orleans.Persistence.AdoNet/Options/AdoNetGrainStorageOptions.cs

```diff
@@ -0,0 +1,138 @@
+        private readonly string name;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="configurationOptions">The option to be validated.</param>
+        /// <param name="name">The name of the option to be validated.</param>
+        public AdoNetGrainStorageOptionsValidator(AdoNetGrainStorageOptions configurationOptions, string name)
+        {
+            if(configurationOptions == null)
+                throw new OrleansConfigurationException($"Invalid AdoNetGrainStorageOptions for AdoNetGrainStorage {name}. Options is required.");
+            this.options = configurationOptions;
+            this.name = name;
+        }
+        /// <inheritdoc cref="IConfigurationValidator"/>
+        public void ValidateConfiguration()
+        {
+            if (string.IsNullOrWhiteSpace(this.options.ConnectionString))
+            {
+                throw new OrleansConfigurationException($"Invalid AdoNetGrainStorageOptions for AdoNetGrainStorage {name}. ConnectionString is required.");
+            }
+            if (options.UseXmlFormat&&options.UseJsonFormat)
+            {
+                throw new OrleansConfigurationException($"Invalid AdoNetGrainStorageOptions for AdoNetGrainStorage {name}. Only one serializer and deserializer should be given.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// Formatter resolver for AdoNetGrainStorageOptions
+    /// </summary>
+    public class AdoNetStorageOptionsFormatterResolver : IOptionFormatterResolver<AdoNetGrainStorageOptions>
+    {
+        private IOptionsSnapshot<AdoNetGrainStorageOptions> optionsSnapshot;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="optionsSnapshot">Option snapshot for AdoNetGrainStorageOptions</param>
+        public AdoNetStorageOptionsFormatterResolver(IOptionsSnapshot<AdoNetGrainStorageOptions> optionsSnapshot)
+        {
+            this.optionsSnapshot = optionsSnapshot;
+        }
+
+        /// <inheritdoc cref="IOptionFormatterResolver"/>
+        public IOptionFormatter<AdoNetGrainStorageOptions> Resolve(string name)
+        {
+            return new AzureTableStorageOptionsFormatter(name, optionsSnapshot.Get(name));
+        }
+
+        private class AzureTableStorageOptionsFormatter : IOptionFormatter<AdoNetGrainStorageOptions>
+        {
+            public string Name { get; }
+
+            private AdoNetGrainStorageOptions options;
+            public AzureTableStorageOptionsFormatter(string name, AdoNetGrainStorageOptions options)
+            {
+                this.options = options;
+                this.Name = OptionFormattingUtilities.Name<AdoNetGrainStorageOptions>(name);
+            }
+
+            public IEnumerable<string> Format()
+            {
+                return new List<string>()
+                {
+                    OptionFormattingUtilities.Format(nameof(this.options.ConnectionString),this.options.ConnectionString),
```

Comment:
This value must be redacted. Use `ConfigUtilities.RedactConnectionStringInfo`

Reply:
We should take steps towards security...

Reply:
Defintely should be redacted, we can -never- log secrets.  However, I'm unsure ConfigUtilities.RedactConnectionStringInfo will redact from an ado connection string. 

@veikkoeeva, are you confident that the connection string format is consistent?

Reply:
we should do a pass on redacting all the connection string which were logged.... I might just do it in this pr

Reply:
I will edit ConfigUtilities.RedactConnectionStringInfo, adding a default logic, which basic replace the connection string with "<--SNIP--> " if none of the secrectKeys are in the conneciton string. 
This should be enough to handle connection string format not consistent problem.

Reply:
And I'm of the opinion on we should just use the default logic, instead of creating complex logic on finding the secret key and just replace the secret key. I don't see much benefits of us doing that.